metricshub-snmp-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/snmp/detection/SnmpGetNextCriterionProcessor.java

```diff
@@ -195,36 +195,54 @@ private static CriterionTestResult checkSNMPGetNextExpectedValue(
 	) {
 		String message;
 		boolean success = true;
-		final Matcher matcher = SNMP_GET_NEXT_VALUE_PATTERN.matcher(result);
-		if (matcher.find()) {
-			final String value = matcher.group(1);
-			final Pattern pattern = Pattern.compile(
-				PslUtils.psl2JavaRegex(expected),
-				Pattern.CASE_INSENSITIVE | Pattern.MULTILINE
-			);
-			if (!pattern.matcher(value).find()) {
+		if (result == null) {
+			message =
+				String.format(
+					"Hostname %s - SNMP test failed - SNMP GetNext of %s was unsuccessful due to a null result.",
+					hostname,
+					oid
+				);
+			success = false;
+		} else if (result.isBlank()) {
```

Comment:
The blank value wasn't regarded as a failure, so allow it to pass through the matcher.